software/controller/lib/debug/trace_cmd.cpp

```diff
@@ -14,110 +14,81 @@
-    VarType::INT32, "trace_var3", "Variable to be saved to the trace buffer",
-    "%d", [] { return trace.GetTracedVarId<2>(); },
-    [](int32_t value) { trace.SetTracedVarId<2>(value); });
-static FnDebugVar vartrace_var4(
-    VarType::INT32, "trace_var4", "Variable to be saved to the trace buffer",
-    "%d", [] { return trace.GetTracedVarId<3>(); },
-    [](int32_t value) { trace.SetTracedVarId<3>(value); });
-
-DbgErrCode TraceCmd::HandleCmd(CmdContext *context) {
+#include "commands.h"
+
+namespace Debug::Command {
+
+ErrorCode TraceHandler::Process(Context *context) {
 
   // The first byte of data is always required, this
   // gives the sub-command.
-  if (context->req_len < 1)
-    return DbgErrCode::MISSING_DATA;
+  if (context->request_length < 1)
+    return ErrorCode::kMissingData;
 
-  switch (context->req[0]) {
+  switch (context->request[0]) {
 
   // Sub-command 0 is used to flush the trace buffer
   // It also disables the trace
   case 0: {
-    trace.SetFlags(0);
-    context->resp_len = 0;
-    return DbgErrCode::OK;
+    trace_->Stop();
+    context->response_length = 0;
+    return ErrorCode::kNone;
   }
 
   // Sub-command 1 is used to read data from the buffer
   case 1:
-    return ReadTraceBuff(context);
+    return ReadTraceBuffer(context);
 
   default:
-    return DbgErrCode::RANGE;
+    return ErrorCode::kInvalidData;
   }
 }
 
-DbgErrCode TraceCmd::ReadTraceBuff(CmdContext *context) {
+ErrorCode TraceHandler::ReadTraceBuffer(Context *context) {
   // See how many active trace variables there are
   // This gives us our sample size;
-  int vct = trace.GetNumActiveVars();
+  uint32_t var_count = trace_->GetNumActiveVars();
 
   // If there aren't any active variables, I'm done
-  if (!vct) {
-    context->resp_len = 0;
-    return DbgErrCode::OK;
+  if (!var_count) {
+    context->response_length = 0;
+    return ErrorCode::kNone;
   }
 
   // See how many samples I can return
   // First, find out how many I could based on the max value
-  int max = context->max_resp_len / static_cast<int>((vct * sizeof(uint32_t)));
+  uint32_t max_samples = context->max_response_length /
+                         static_cast<uint32_t>(var_count * sizeof(uint32_t));
 
-  // If there's not room for even one sample, return an error.
+  // If there's not enough room for even one sample, return an error.
   // That really shouldn't happen
-  if (!max)
-    return DbgErrCode::NO_MEMORY;
+  if (max_samples == 0) {
+    return ErrorCode::kNoMemory;
+  }
 
   // Find the total number of samples in the buffer
-  int tot = trace.GetNumSamples();
-  if (tot > max)
-    tot = max;
+  uint32_t samples_count = trace_->GetNumSamples();
+  if (samples_count > max_samples) {
+    samples_count = max_samples;
+  }
 
-  uint8_t *resp = context->resp;
-  std::array<uint32_t, TRACE_VAR_CT> record;
-  for (int i = 0; i < tot; i++) {
+  std::array<uint32_t, kMaxTraceVars> record;
+  for (uint32_t sample = 0; sample < samples_count; sample++) {
```

Comment:
`size_t`? And the few above this as well?

Reply:
in this case yes.